marklogic-data-hub-central/ui/src/components/entities/mapping/create-edit-mapping-dialog/create-edit-mapping-dialog.tsx

```diff
@@ -17,11 +17,14 @@ const CreateEditMappingDialog = (props) => {
   const [selectedSource, setSelectedSource] = useState(props.mapData && props.mapData != {} ? props.mapData.selectedSource : 'collection')
   const [srcQuery, setSrcQuery] = useState(props.mapData && props.mapData != {} ? props.mapData.sourceQuery : '');
   const [isQuerySelected, setIsQuerySelected] = useState(false);
+
   //To check submit validity
   const [isMapNameTouched, setMapNameTouched] = useState(false);
+  const [isNameMissingOnSave, setNameMissingOnSave] = useState(false);
```

Comment:
Seems like we already have a boolean flag for mapName as "isMapNameTouched" which we can leverage to see of mapName has ever been modified or not. Same goes for other mandatory fields in the form.
I would suggest to just utilize the required field values and the corresponding '...touched' fields to get the desired output instead of creating additional flags.

Reply:
Fixed!